clippy_lints/src/matches/manual_unwrap_or.rs

```diff
@@ -62,11 +62,11 @@ fn get_some_and_none_bodies<'tcx>(
     arm2: &'tcx Arm<'tcx>,
 ) -> Option<((&'tcx Expr<'tcx>, HirId), &'tcx Expr<'tcx>)> {
     if let Some(binding_id) = get_some(cx, arm1.pat)
-        && let Some(body_none) = get_none(cx, arm2)
+        && let Some(body_none) = get_none(cx, arm2, true)
     {
         Some(((arm1.body, binding_id), body_none))
     } else if let Some(binding_id) = get_some(cx, arm2.pat)
-        && let Some(body_none) = get_none(cx, arm1)
+        && let Some(body_none) = get_none(cx, arm1, false)
```

Comment:
Having these two comparisons just change by a single character is very unclear, and with the `false` and `true` even more so! Could you change the order so that we always check for `arm1` above and `arm2` below?


```suggestion
} else if let Some(binding_none) = get_none(cx, arm1, false)
        && let Some(body_id) = get_some(cx, arm2.pat)
```